### PR TITLE
Improve connection handling on suspend/wakeup.

### PR DESCRIPTION
**Pull request by ksooo**
This is an old patch I have on my private build for about a year now.

Today, on system suspend, pvr.hts looses connection to tvh (socket closed) and immediately retries to reconnect, even though suspend is in progress and the attempt cannot succeed. This results in error log messages while suspending and a non-deterministic behaviour after resume. This PR fixes this problem by preventing reconnects as long as pvr.hts is "suspended".

@Jalle19 objections?

### CODE REVIEW

File: src/HTSPConnection.cpp [hunk 1]
```diff
@@ -495,10 +515,9 @@ void CHTSPConnection::Register ( void )
     return;
   }
 
-  /* Rate limit retry */
 fail:
-  Sleep(5000);
```
Comment (Jalle19) [inline src/HTSPConnection.cpp]: Is this removal intentional?

**Comment by Jalle19** [comment]
Looks good in general, any improvement to wake/suspend handling is much appreciated.

File: src/HTSPConnection.cpp (hunk 1 above)
Comment (ksooo) [inline src/HTSPConnection.cpp]: Yes.

**Comment by ksooo** [comment]
Okay then. This on was (most probably ) my last contribution for the next 2 weeks. Vacation time. ;-)

**Comment by Jalle19** [comment]
Have a nice vacation, you've definitely earned it!
